vllm/platforms/tpu.py

```diff
@@ -110,20 +110,21 @@ def inference_mode(cls):
 
     @classmethod
     def check_and_update_config(cls, vllm_config: VllmConfig) -> None:
-        from vllm.config import CompilationLevel, CUDAGraphMode
+        from vllm.config import CompilationMode, CUDAGraphMode
 
         cache_config = vllm_config.cache_config
         # For v0, the default block size is 16.
         if cache_config and cache_config.block_size is None:
             cache_config.block_size = cast(BlockSize, 16)
         compilation_config = vllm_config.compilation_config
 
-        # TPU only supports DYNAMO_ONCE compilation level
-        if compilation_config.level != CompilationLevel.DYNAMO_ONCE:
+        # TPU only supports DYNAMO_TRACE_ONCE compilation level
```

Comment:
This is ok because TPU requires compilation and doesn't use inductor at all. But you can ask in #sig-tpu in slack if VLLM_COMPILE is completely off the table, it might not be? At least in the long run